### ISSUE TITLE
ibc-hooks: failing unit tests when using with ibc-go >= 7.1.0

### ISSUE THREAD

**Issue by xlab** [cosmos/ibc-apps#57]
Hello! Recent changes in [ibc-go 7.1.0](https://github.com/cosmos/ibc-go/releases/tag/v7.1.0) (see **apps/transfer** in release notes and specifically [ADR-011](https://github.com/cosmos/ibc-go/blob/v7.1.0/docs/architecture/adr-011-transfer-total-escrow-state-entry.md) for the background) are breaking unit tests for `ibc-hooks` package. While it can still work fine with `ibc-go 7.2.0`, this mock of escrow in the test is not sufficient anymore:
https://github.com/cosmos/ibc-apps/blob/a1e480497c1230e012ffc37777e3b2265c4ac2a9/modules/ibc-hooks/tests/unit/module_test.go#L103-L105

Will result in a panic like this:
```
receiver cosmos1axvk83fmkhz98wqzse6ds0aq53q99xs7cc0478yungfeflscex0qlg7stl unescrows 1stake
    suite.go:87: test panicked: negative coin amount: -1stake
        goroutine 12 [running]:
        runtime/debug.Stack()
        	/usr/local/go/src/runtime/debug/stack.go:24 +0x64
```

This is because escrow amounts now tracked in TransferKeeper as well, not only as balances in the bank module.

To properly simulate escrowing tokens without actually sending them, the caller must also set the total escrow amount in the TransferKeeper directly:
```go
// SetTotalEscrowForDenom stores the total amount of source chain tokens that are in escrow.
// Amount is stored in state if and only if it is not equal to zero. The function will panic
// if the amount is negative.
func (k Keeper) SetTotalEscrowForDenom(ctx sdk.Context, coin sdk.Coin) 
```

I have made necessary changes in PR https://github.com/cosmos/ibc-apps/pull/56 please note that it supports versions prior to `7.1.0` as well, that don't have `SetTotalEscrowForDenom` available in TransferKeeper.
```[tasklist]
### Tasks
- [ ] https://github.com/cosmos/ibc-apps/pull/56
```

**Comment by akc2267** [comment]
This needs to be assigned to someone from Osmosis

**Comment by jtieri** [comment]
cc @nicolaslara

**Comment by nicolaslara** [comment]
That makes a lot of sense. Thanks for the fix! (approved the PR)

**Comment by jtieri** [comment]
Closed by #56
